cmd/puppeth/ssh.go

```diff
@@ -77,7 +77,23 @@ func dial(server string, pubkey []byte) (*sshClient, error) {
 		log.Warn("No SSH key, falling back to passwords", "path", path, "err", err)
 	} else {
 		key, err := ssh.ParsePrivateKey(buf)
-		if err != nil {
+		if string(err.Error()) == "ssh: cannot decode encrypted private keys" {
```

Comment:
You don't need the string case here, `Error()` already returns a string.

Also this is a very brittle way to check for the error as it could change at any moment, however I do notice that the crypto package does not have an expose error constant for this. I'd suggest leaving this check as you implemented for now, but opening a PR against the upstream repo to make the error a constant.